migrations/20180325211342_migrate-posts.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable */
+exports.up = async knex => {
+  // Create the new table where relevant posts and posts_meta will go in
+  await knex.schema.createTable('articles', table => {
+    table.increments('id').primary().unique().notNullable();
+    table.string('slug', 150).unique().notNullable();
+    table.string('title', 150).notNullable();
+    table.text('markdown', 'mediumtext');
+    table.text('html', 'mediumtext');
+    table.string('image_url');
+    table.string('teaser');
+    table.integer('views').unsigned().notNullable().defaultTo(0);
+    table.dateTime('published_at');
+    table.boolean('is_interactive').notNullable().defaultTo(false);
+    table.integer('category_id').unsigned().references('id').inTable('categories')
+      .onDelete('SET NULL').onUpdate('CASCADE');
+  });
+  // Now we copy over the data from the two tables into the new one
+
+  // We first have to convert the posts table to latin1 so we correctly interpret it
+  const tableName = 'posts';
+  const charset = 'latin1';
+  await knex.schema.raw(`ALTER TABLE ${tableName} CONVERT TO CHARACTER SET ${charset}`);
+
+  const postRows = await knex.select('*').from('posts');
+  const metaRows = await knex.select('*').from('posts_meta');
+
+  const convertedPostRows = postRows.map(row => ({
+    id: row.id,
+    title: row.title,
+    slug: row.slug,
+    markdown: row.markdown,
+    html: row.html,
+    image_url: row.image,
+    teaser: row.meta_description,
+  }));
+  const convertedMetaRows = metaRows.map(row => ({
+    id: row.id,
+    category_id: row.category_id,
+    views: row.views,
+    published_at: row.gazelle_published_at,
+    is_interactive: row.is_interactive,
+  }));
+
+  const articleRows = convertedPostRows.map(postRow => {
+    const metaRow = convertedMetaRows.find(_metaRow => _metaRow.id === postRow.id) || {};
+    return {
+      ...postRow,
+      ...metaRow,
+    };
+  }).filter(row => row);
```

Comment:
Is this filter needed? It doesn't seem to do anything :)

Reply:
Good catch! I had that on at a point where I was only migrating the posts that we already had Gazelle meta data on and returning null if there wasn't any, but it created some foreign key issues so now I'm just migrating everything